File: app/src/main/java/com/example/readative/room/entity/Language.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Language {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "language_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "language_name")
    public String name;
    @NonNull
    @ColumnInfo(name = "language_code")
    public String code;
    @NonNull
    @ColumnInfo(name = "language_directionality")
    public String directionality;
    @NonNull
    @ColumnInfo(name = "language_default_font")
    public String defaultFont;
}

File: app/src/main/java/com/example/readative/persistance/entities/Language.kt
package com.example.readative.persistance.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "languages",
    indices = [
        Index(value = ["name"], unique = true),
        Index(value = ["code"], unique = true)
    ]
)
data class Language(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "code") val code: String,
    @ColumnInfo(name = "directionality") val directionality: LanguageDirectionality = LanguageDirectionality.LEFT_TO_RIGHT,
    @ColumnInfo(name = "default_font") val default_font: Int,
) : ReadativeEntity
